starfish/imagestack/imagestack.py

```diff
@@ -224,22 +225,31 @@ def __repr__(self):
         return f"<starfish.ImageStack ({shape})>"
 
     @classmethod
-    def from_tileset(cls, tileset: TileSet) -> "ImageStack":
+    def from_tileset(
+            cls,
+            tileset: TileSet,
+            crop_parameters: Optional[CropParameters]=None,
+    ) -> "ImageStack":
         """
         Parse a :py:class:`slicedimage.TileSet` into an ImageStack.
 
         Parameters:
         -----------
         tileset : TileSet
             The tileset to parse.
+        crop_parameters : Optional[CropParameters]
+
 
         Returns:
         --------
         ImageStack :
             An ImageStack representing encapsulating the data from the TileSet.
         """
-        parsed = parse_tileset(tileset)
-        return cls(*parsed)
+        tile_shape, tile_data = parse_tileset(tileset)
+        if crop_parameters is not None:
+            tile_shape = crop_parameters.crop_shape(tile_shape)
```

Comment:
Do you need to do the cropping of the shape independently of the image cropping? Couldn't you just crop the TileCollection then derive the shape from the result?

Reply:
Yes, one could do that, but `CroppingTileCollectionData` doesn't know its shape.  One would need to fetch a tile and grab its shape to know (kind of duplicating the logic in `starfish.imagestack.parser.tileset._parser.parse_tileset`